llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -754,6 +754,17 @@ InstructionCost VPInstruction::computeCost(ElementCount VF,
     return Cost + Ctx.TTI.getVectorInstrCost(Instruction::ExtractElement, VecTy,
                                              Ctx.CostKind);
   }
+  case VPInstruction::FirstOrderRecurrenceSplice: {
+    assert(VF.isVector() && "Scalar FirstOrderRecurrenceSplice?");
+    SmallVector<int> Mask(VF.getKnownMinValue());
+    std::iota(Mask.begin(), Mask.end(), VF.getKnownMinValue() - 1);
+    Type *VectorTy =
+        toVectorTy(Ctx.Types.inferScalarType(this->getVPSingleValue()), VF);
```

Comment:
```suggestion
        toVectorTy(Ctx.Types.inferScalarType(this()), VF);
```

Is `getVPSingleValue` needed? VPInstruction is a `VPValue`

Reply:
Looks like it's not, I've removed it